string/location.ts

```diff
@@ -0,0 +1,21 @@
+import { TextLocation, TextWithLocation } from "./types.ts";
+
+export function location(text: string, index: number): TextLocation {
+  const linesText = text.slice(0, index).split("\n");
+
+  const line = linesText.length;
+  const column = linesText[linesText.length - 1].length + 1;
+
+  const location = { offset: index, line: line, column: column };
+  return location;
+}
+
+export function locateRegex(text: string, regex: RegExp): TextWithLocation[] {
+  const matches = [...text.matchAll(regex)];
+
+  return matches.map((thisMatch) => {
+    const match = thisMatch[0];
+
+    return [match, location(text, thisMatch.index ?? 0)];
+  });
```

Comment:
This can be boiled down to:

```ts
return [...text.matchAll(regex)]
  .map((match) => [match[0], location(text, thisMatch.index ?? 0)])
```

This means that it's just as many lines to write out by hand as it is to import (1 line) and use (a 2nd line) this utility function, so it saves the user very little. It's closer to a snippet than a utility function... actually made me think we should consider creating snippets. Created #80 to track.

Anyway, I wouldn't recommend creating a util for this, but if you strongly disagree I won't block the PR over it 👍

Reply:
Agreed. Great point. Removed this and will consider a snippet later.